testdata/cheats/Wallet.t.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "./Cheats.sol";
+
+contract Foo {}
+
+contract WalletTest is DSTest {
+    Cheats constant cheats = Cheats(HEVM_ADDRESS);
+
+    function testCreateWalletStringPrivAndLabel() public {
+        bytes memory privKey = "this is a priv key";
+        Cheats.Wallet memory wallet = cheats.createWallet(string(privKey));
+
+        address expectedAddr = cheats.addr(wallet.privateKey);
+        assertEq(expectedAddr, wallet.addr);
+
+        string memory label = cheats.getLabel(wallet.addr);
+        assertEq(label, string(privKey), "labelled address != wallet.addr");
+    }
```

Comment:
Ideally, there'd be another section of this test that checks the validity of `wallet.publicKeyX` and `wallet.publicKeyY`, but I wasn't sure what the best way of doing that was

Reply:
One way would be to verify the address derived from the coordinates equals the address derived via the `addr(unit privKey)(address)` cheatcode.

Deriving the Ethereum address is specified in the Yellow Paper "Appendix F: Signing Transactions" §134. A possible Solidity implementation is:
```solidity
function addressOf(uint xCoordinate, uint yCoordinate) internal pure returns (address) {
    return address(uint160(keccak256(abi.encode(xCoordinate, yCoordinate))));
}
```